ios/CodePush/CodePush.m

```diff
@@ -455,9 +455,12 @@ - (void)rollbackPackage
 {
     NSError *error;
     NSDictionary *failedPackage = [CodePushPackage getCurrentPackage:&error];
-
-    // Write the current package's metadata to the "failed list"
-    [self saveFailedUpdate:failedPackage];
+    if (error) {
```

Comment:
Anyway, I don't see what change you are suggesting either. Are you saying we should remove this error pointer check save the `failedPackage` even if we find that the error pointer is set? Because that doesn't sound safe to me at all. I would rather take the safe route and not write the `failedPackage` if we detect either of the failure conditions.